src/software/ai/evaluation/calc_best_shot.h

```diff
@@ -181,4 +182,32 @@ namespace Evaluation
     double calcShotOpenEnemyNetPercentage(const Field &field, const Point &shot_origin,
                                           const Shot &shot);
 
+    /**
+     * Function calculates the optimal shot location and the corresponding Angle
+     * representing the 'open' area of that shot at the segment
+     *
+     * @param origin : The origin of the shot
+     * @param segment : The segment at which shots are being evaluated on
+     * @param obstacles : Any obstacle that can block the shot
+     *
+     * @return Shot : Returns the optimal Shot (Point and Angle) corresponding to the
+     * given parameters
+     */
+    Shot calcMostOpenDirection(Point origin, Segment segment,
+                               std::vector<Circle> obstacles);
+
+    /**
+     * Function calculates the optimal shot location and the corresponding Angle
+     * representing the 'open' area of that shot at the segment
+     *
+     * @param origin : The origin of the shot
+     * @param segment : The segment at which shots are being evaluated on
+     * @param robot_obstacles : Any Robot (friendly or enemy) that can block the shot
+     *
+     * @return Shot : Returns the optimal Shot (Point and Angle) corresponding to the
+     * given parameters
+     */
+    Shot calcMostOpenDirection(Point origin, Segment segment,
```

Comment:
Is it this being used anywhere? If so, can we rename them to make it easier to differentiate between the two `calcMostOpenDirection`? I'm having a hard time code reading these two overloaded functions

Reply:
Renamed to differentiate the two functions